cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -353,6 +354,26 @@ func StartControllers(s *options.CMServer, kubeClient *client.Client, kubeconfig
 		serviceaccountcontroller.DefaultServiceAccountsControllerOptions(),
 	).Run()
 
+	if s.HealthCheckControllerSyncPeriod > 0 {
+		config := client.AddUserAgent(kubeconfig, "healthchecker-controller")
+
+		// NB: in order for this health checker to work, the controller manager must be
+		// able to reach all of the API servers listed in the "default/kubernetes"
+		// endpoints.
+		checkerTransport, err := client.TransportFor(config)
+		if err != nil {
+			panic(err)
```

Comment:
I agree about not doing anything silently. I was suggesting continue with a warning and not failing 
It seems to me that the main controller works if all sub controller work and fails even if one fails. If that is desired/or by design, that's fine with me.